dhall-docs/src/Dhall/Docs/CodeRenderer.hs

```diff
@@ -68,12 +77,25 @@ getSourceLine, getSourceColumn :: SourcePos -> Int
 getSourceLine = SourcePos.unPos . SourcePos.sourceLine
 getSourceColumn = SourcePos.unPos . SourcePos.sourceColumn
 
+-- | To support jump-to-definition on record literals we need to know the
+--   the type of a name as we traverse the AST. Using 'Dhall.TypeCheck' is not
+--   possible since we always try to extract as much information as we can from
+--   source.
+data DhallType
```

Comment:
I was initially confused in what way this represents a "type". I wonder whether something like `JtdInfo` or maybe `Provenience` might be slightly less confusing.

Reply:
I'll use `JtdInfo`. `Provenience` doesn't seem the best term from my POV.